File: assets/code-variants/ch12/26.java
private NameService nameService;
private UserStore userStore;

@Before
public void setUp() {
    nameService = new NameService();
    nameService.set("user1", "Donald Knuth");
    userStore = new UserStore(nameService);
}

@Test
public void shouldReturnNameFromService() {
    nameService.set("user1", "Margaret Hamilton");
    UserDetails user = userStore.get("user1");
    assertThat(user.getName()).isEqualTo("Margaret Hamilton");
}

File: assets/code-variants/ch12/26.py
class UserStoreTest(unittest.TestCase):
    def setUp(self):
        self.name_service = NameService()
        self.name_service.set("user1", "Donald Knuth")
        self.user_store = UserStore(self.name_service)

    def test_should_return_name_from_service(self):
        self.name_service.set("user1", "Margaret Hamilton")
        user = self.user_store.get("user1")
        self.assertEqual(user.name, "Margaret Hamilton")
